### PR TITLE
feat: replace symmetric encryption algorithm by aes-256-gcm

### PR DESCRIPTION
**Pull request by vrolland**
## Description of the changes

Following recommendations in the findings "RN-01-006 WP1" and "RN-01-007 WP1" of the encryption audit by Cure+53

### CODE REVIEW

File: packages/utils/src/crypto/crypto-wrapper.ts [hunk 1]
```diff
@@ -47,6 +52,28 @@ async function encryptWithAes256cbc(data: Buffer, key: Buffer): Promise<Buffer>
...
+ * Encrypts a buffer using AES-256-gcm plus a random Initialization Vector (IV)
+ *
+ * @param data the data to encrypt
+ * @param key the key that will be used for the encryption
+ *
+ * @returns Promise resolving a buffer containing the IV and the encrypted data
+ */
+async function encryptWithAes256gcm(data: Buffer, key: Buffer): Promise<Buffer> {
+  // Generate randomly the Initialization Vector
+  const iv = await randomBytes(INITIALIZATION_VECTOR_LENGTH);
```
Comment (rittme) [inline packages/utils/src/crypto/crypto-wrapper.ts:65]: Is this random enough?

File: packages/utils/src/crypto/crypto-wrapper.ts [hunk 2]
```diff
@@ -58,11 +85,41 @@ async function encryptWithAes256cbc(data: Buffer, key: Buffer): Promise<Buffer>
 async function decryptWithAes256cbc(encryptedAndIv: Buffer, key: Buffer): Promise<Buffer> {
   // Get the IV
   const iv = encryptedAndIv.slice(0, INITIALIZATION_VECTOR_LENGTH);
+
   // Get the encrypted data itself
   const encryptedData = encryptedAndIv.slice(INITIALIZATION_VECTOR_LENGTH);
 
   // Create the decipher object
   const decipher = createDecipheriv(AES_256_CBC_ALGORITHM, key, iv);
+  // decipher.setAuthTag(authTag);
```
Comment (rittme) [inline packages/utils/src/crypto/crypto-wrapper.ts:94]: Forgot to remove?

File: packages/utils/src/crypto/crypto-wrapper.ts (hunk 1 above)
Comment (rittme) [reply packages/utils/src/crypto/crypto-wrapper.ts:65]: Actually answered in the other PR :)

File: packages/utils/src/crypto/crypto-wrapper.ts (hunk 2 above)
Comment (vrolland) [reply packages/utils/src/crypto/crypto-wrapper.ts:94]: Yep exactly

**Comment by vrolland** [comment]
> This looks sleek. I love when good design pays off and something that could be a big undertaking is just a few changed files like this slightly_smiling_face

I can't agree more!

**Pull request merged**